### DISCUSSION TITLE
asyncio wrapper for external API using httpx

### DISCUSSION THREAD

**Discussion by umax** [encode/httpx#1649] [Q&A] [answered]
Hi, guys!

I want to use httpx as a base for my client to external API. I want to create wrapper around external API as a module with several methods. The question is: what is the right way to use async httpx client? option 1 or 2?

### Option1 (use global session)
mywrapper.py
```python
_client = httpx.AsyncClient()

async def method1():
    await _client.METHOD()

async def method2():
    await _client.METHOD()
```

### Option2 (create new session each time)
mywrapper.py
```python
async def method1():
    async with httpx.AsyncClient() as client:
        await client.METHOD()

async def method2():
    async with httpx.AsyncClient() as client:
        await client.METHOD()
```

Thank you!

**Comment by johtso** [comment, marked as answer]
You might be interested in this discussion: https://github.com/encode/httpx/discussions/1552

**Reply by umax** [reply]
That's what I'm looking for. Thank you @johtso !
